src/js/components/DataSearch/DataSearch.js

```diff
@@ -18,8 +18,29 @@ const dropProps = {
   align: { top: 'bottom', left: 'left' },
 };
 
+// 300ms was chosen empirically as a reasonable default
+const DEBOUNCE_TIMEOUT = 300;
```

Comment:
Instead of repeating some of the debounce functions in DataSearch + DataForm, can we have the logic live in a shared util file and import them into each component accordingly?

Reply:
makes sense.